Source/Libraries/Adapters/MySqlAdapters/MySqlOutputAdapter.cs

```diff
@@ -176,17 +176,12 @@ protected override void ProcessMeasurements(IMeasurement[] measurements)
                 foreach (IMeasurement measurement in measurements)
                 {
                     // Create the command string to insert the measurement as a record in the table.
-                    StringBuilder commandString = new StringBuilder("INSERT INTO Measurement VALUES ('");
                     IDbCommand command = m_connection.CreateCommand();
+                    command.Parameters.Add(measurement.ID);
+                    command.Parameters.Add((long)measurement.Timestamp);
+                    command.Parameters.Add(measurement.AdjustedValue);
```

Comment:
I have a feeling this won't work. `IDbCommand.Parameters` returns a collection of type `IDataParameterCollection` which implements the non-generic `IList` interface. It's entirely possible that the MySQL implementation of the `Add()` method can convert any old value to an instance of `IDbDataParameter`, but I feel it's pretty unlikely. You'll need to use `command.CreateParameter()` and then set the `ParameterName` and `Value` fields for each parameter.